### ISSUE TITLE
Dashboard Variables: Current value saved even when variable throws an error

### ISSUE THREAD

**Issue by jcolladokuri** [grafana/grafana#98296]
### What happened?

After a dependent variable loads with values and something is selected, if I change the value of one of the preceding variables that results in an error in one or more of the child variables, the last known good value is still persisted in the dashboard JSON.

Loaded variables with no errors:

![Image](https://github.com/user-attachments/assets/536448aa-0a6d-43b7-9414-be9a611f7225)

Persisted in JSON as expected:

![Image](https://github.com/user-attachments/assets/34d8cc5a-a9be-41ef-8de8-e29d36053c73)

Switch a parent variable to something that will throw an error (in this case it is an access error):

![Image](https://github.com/user-attachments/assets/876e5554-35ee-440f-9cd3-7ec8b6ae24dc)

Notice how even with a different subscription, I still get the same value back for "Resource" when it should have been empty, as that resource does not exist in the selected subscription variable. I expect resources to clear and show None, the same way Resource Group does. Additionally, it is not just the selected value that is persistent, but all the options in the dropdown.

Wrong value persisted in the Dashboard JSON:

![Image](https://github.com/user-attachments/assets/d0fc295f-7df9-4ee7-aa8b-062584776562)

With this behavior, it leads to innacuracy when using `Save current variable values as dashboard default` as across Saves and Loads it will continue to persist a wrong value for the variable that should have ideally been empty.

### What did you expect to happen?

Whenever there is an error, or no options are returned for a specific "child" variable, the value of those variables should be set to none, rather than the previous last known good value for that variable.

### Did this work before?

Unable to tell, I was able to repro in 10.4.1 as well

### How do we reproduce it?

1. Create any number of chained variables.
2. change the value to one of the parents to something that will force an error in the dependent chained variables
3. Look at dashboard JSON and search for any "current" selected variable value that should have been empty.
4. Save the dashboards with the current selection for variables and notice that because the variable value was not cleared previously, it is now saved for the next time the dashboard is loaded.

### Environment (with versions)?

Grafana: 11.2.2
OS: Windows
Browser: Edge

### Grafana platform?

None

### Datasource(s)?

For this example, Azure Monitor template variables were used.

**Comment by kayodeprinceMS** [comment]
cc: @taleena for routing.

For clarity, this is behavior is not limited to the Azure Monitor data source. It is also reproducible with query variables from other data sources. The `Save current variable values as dashboard default` option saves the last known good value into the dashboard JSON even in cases where the current and most recent variable query results in an error and the value should be empty/null. 

When the dashboard is re-opened the older last known (stale) value is loaded from the JSON instead of the variable queries being re-run and the (valid) empty/null value loaded from the failed query.

**Comment by taleena** [comment]
@jcolladokuri given @kayodeprinceMS comment, where subscriptions always included, even when you were testing other datasouces (and which others did you test - I saw geneva there, at least)?

**Comment by kayodeprinceMS** [comment]
No, not limited to subscriptions. The original customer reported issue was in ADX and we saw the same behavior in Azure Monitor as shown in the screenshots above.

**Comment by taleena** [comment]
cc @tdbishop

**Comment by changov** [comment]
The persistence of stale and incoherent variable values, in particular for hidden variables, keeps causing massive confusion for our users, and we have no means of detecting this situation in the dashboard's logic. As Kayode suggested, the desired fix is to clear all dependent variables when re-evaluation of a parent variable fails. This leaves dashboards authors with the generic responsibility of making queries work correctly (or fail properly and explicitly) when not all required variables have values selected.

**Comment by hugohaggmark** [comment]
Changing how variables behave would add significant complexity, so instead we’re implementing a warning that appears when a user tries to save a dashboard containing variables with errors. Details are in PR #110154.
